### PR TITLE
Swift 3.1

### PR DESCRIPTION
**Pull request by wisolith**
Migrate to Swift 3.1

### CODE REVIEW

File: Scenarios/StepArguments.swift [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
-    self.captures = matchResult.captures
+    self.captures = matchResult.captures.flatMap { $0 }
   }
 
   public subscript(index: Int) -> String {
     return captures[index]
   }
 
+  public func index(after i: Int) -> Int {
+    return i + 1
```
Comment (gavanchan-outware) [inline Scenarios/StepArguments.swift]: Following the implementation of 'subscript', I would prefer forwarding this call to the underlying 'captures' collection, to retain the collection's semantics. Is this explicitly `i + 1`?

**Review by gavanchan-outware — CHANGES_REQUESTED** [review]
Looks good! Let me know what you think, and let's get this through! 🙂

**Review by gavanchan-outware — CHANGES_REQUESTED** [review]
Beyond the codebase:

- Travis seems to be using an outdated version of Xcode. The Travis YAML file will need to be updated to use the required version of Xcode.
- This branch is out-of-sync with the latest branch on the main repository. Could you attempt a sync with the main repository?
- Nimble v7.0.0 does not behave as expected, in some cases. Do you feel that it would be good to revert to using the previous major version?

File: Scenarios/StepArguments.swift (hunk 1 above)
Comment (wisolith) [reply Scenarios/StepArguments.swift]: Something like `return captures.index(after: i)`?

**Comment by wisolith** [comment]
- Updated `.travis.yml` to Xcode 8.3
- Rebased onto the current `master` of `outware/Scenarios`
- Set Nimble version to `5.1`

File: Scenarios/StepArguments.swift (hunk 1 above)
Comment (gavanchan-outware) [reply Scenarios/StepArguments.swift]: Yeah, if the implementation isn't meant to be explicitly `i + 1`, I think that would be more representative of the type.

**Review by gavanchan-outware — APPROVED** [review]

**Comment by gavanchan-outware** [comment]
Looks good! Passes on Travis and when run on my local machine using Xcode 8.3.

Thanks for that, @wisolith!

Time to merge!

**Pull request merged**